unstructured_inference/inference/layoutelement.py

```diff
@@ -301,8 +306,9 @@ def merge_text_regions(regions: List[TextRegion]) -> TextRegion:
     max_y2 = max([tr.y2 for tr in regions])
 
     merged_text = " ".join([tr.text for tr in regions if tr.text])
-
-    return TextRegion(min_x1, min_y1, max_x2, max_y2, merged_text)
+    sources = [*{tr.source for tr in regions}]
+    source = sources.pop() if len(sources) == 1 else "merged:".join(sources)  # type:ignore
```

Comment:
the string "merged" should probably be part of the class attribute for `TextRegion` or defined somewhere that can be imported so that downstream can know what joiner was used to join the list. Also you can just say `"merged:".join(sources)`. If the list has only one member it will just return itself, the same as `souces.pop()`